src/compas_ui/rhino/objects/networkobject.py

```diff
@@ -0,0 +1,344 @@
+        The scale is applied relative to the location of the object in the scene.
+        """
+        if not self._scale:
+            self._scale = 1.0
+        return self._scale
+
+    @scale.setter
+    def scale(self, scale):
+        self._scale = scale
+
+    @property
+    def rotation(self):
+        """list of float:
+        The rotation angles around the 3 axis of the coordinate system
+        with the origin placed at the location of the object in the scene.
+        """
+        if not self._rotation:
+            self._rotation = [0, 0, 0]
+        return self._rotation
+
+    @rotation.setter
+    def rotation(self, rotation):
+        self._rotation = rotation
+
+    @property
+    def node_xyz(self):
+        """dict : The view coordinates of the mesh object."""
+        origin = Point(0, 0, 0)
+        stack = []
+        if self.scale != 1.0:
+            S = Scale.from_factors([self.scale] * 3)
+            stack.append(S)
+        if self.rotation != [0, 0, 0]:
+            R = Rotation.from_euler_angles(self.rotation)
+            stack.append(R)
+        if self.location != origin:
+            if self.anchor is not None:
+                xyz = self.network.vertex_attributes(self.anchor, 'xyz')
+                point = Point(* xyz)
+                T1 = Translation.from_vector(origin - point)
+                stack.insert(0, T1)
+            T2 = Translation.from_vector(self.location)
+            stack.append(T2)
+        if stack:
+            X = reduce(mul, stack[::-1])
+            network = self.network.transformed(X)
+        else:
+            network = self.network
+        node_xyz = {network: network.node_attributes(node, 'xyz') for node in network.nodes()}
+        return node_xyz
+
+    @property
+    def guid_node(self):
+        """dict: Map between Rhino object GUIDs and network node identifiers."""
+        if not self._guid_node:
+            self._guid_node = {}
+        return self._guid_node
+
+    @guid_node.setter
+    def guid_node(self, values):
+        self._guid_node = dict(values)
+
+    @property
+    def guid_edge(self):
+        """dict: Map between Rhino object GUIDs and network edge identifiers."""
+        if not self._guid_edge:
+            self._guid_edge = {}
+        return self._guid_edge
+
+    @guid_edge.setter
+    def guid_edge(self, values):
+        self._guid_edge = dict(values)
+
+    @property
+    def guid_nodelabel(self):
+        """dict: Map between Rhino object GUIDs and network nodelabel identifiers."""
+        return self._guid_vertexlabel
+
+    @guid_nodelabel.setter
+    def guid_nodelabel(self, values):
+        self._guid_vertexlabel = dict(values)
+
+    @property
+    def guid_edgelabel(self):
+        """dict: Map between Rhino object GUIDs and network edgelabel identifiers."""
+        return self._guid_edgelabel
+
+    @guid_edgelabel.setter
+    def guid_edgelabel(self, values):
+        self._guid_edgelabel = dict(values)
```

Comment:
if i remember correctly this was there to allow selection of nodes via their labels. i don't think we should provide that option anymore...